baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/CollectionsIsEmpty.java

```diff
@@ -0,0 +1,120 @@
+import com.google.errorprone.BugPattern.SeverityLevel;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.matchers.method.MethodMatchers;
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.BinaryTree;
+import com.sun.source.tree.ExpressionTree;
+import com.sun.source.tree.LiteralTree;
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree.Kind;
+import java.util.Collection;
+import java.util.Optional;
+import org.immutables.value.Value.Immutable;
+
+@AutoService(BugChecker.class)
+@BugPattern(
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        severity = SeverityLevel.WARNING,
+        summary = "Use the isEmpty method instead of checking collection size")
+public final class CollectionsIsEmpty extends BugChecker implements BugChecker.BinaryTreeMatcher {
+    private static final Matcher<ExpressionTree> COLLECTION_SIZE_METHOD_MATCHER = MethodMatchers.instanceMethod()
+            .onDescendantOf(Collection.class.getName())
+            .named("size");
+
+    @Override
+    public Description matchBinary(BinaryTree tree, VisitorState state) {
+        Optional<EqualsZeroExpression> maybeEqualsZeroExpression = getEqualsZeroExpression(tree);
+        if (maybeEqualsZeroExpression.isEmpty()) {
+            return Description.NO_MATCH;
+        }
+        EqualsZeroExpression equalsZeroExpression = maybeEqualsZeroExpression.get();
+        ExpressionTree operand = equalsZeroExpression.operand();
+
+        if (!(operand instanceof MethodInvocationTree) || !COLLECTION_SIZE_METHOD_MATCHER.matches(operand, state)) {
+            return Description.NO_MATCH;
+        }
+
+        return describeMatch(
+                tree,
+                SuggestedFix.replace(
+                        tree,
+                        (equalsZeroExpression.type() == ExpressionType.NEQ ? "!" : "")
+                                + state.getSourceForNode(ASTHelpers.getReceiver(operand)) + ".isEmpty()"));
+    }
+
+    private static Optional<EqualsZeroExpression> getEqualsZeroExpression(BinaryTree tree) {
+        ExpressionType ret;
+        switch (tree.getKind()) {
+            case EQUAL_TO:
+                ret = ExpressionType.EQ;
+                break;
+            case NOT_EQUAL_TO:
+                ret = ExpressionType.NEQ;
+                break;
+            default:
+                return Optional.empty();
+        }
+        ExpressionTree leftOperand = tree.getLeftOperand();
+        ExpressionTree rightOperand = tree.getRightOperand();
+
+        if (leftOperand.getKind() == Kind.INT_LITERAL
+                && ((LiteralTree) leftOperand).getValue().equals(0)) {
```

Comment:
We could create a `Matchers.intLiteral(0)` to use for these checks (requires the VisitorState to be passed into this method, but may simplify the implementation a bit)